mtt/src/mtt_types.rs

```diff
@@ -118,3 +116,226 @@ impl MttL1EntryType {
         }
     }
 }
+
+// Bits 45:26
+const L2_ADDR_MASK: u64 = 0x3fff_fc00_0000;
+// Shifted right by 26, then 3-bit left shift to account for the 8-byte entry
+const L2_ADDR_SHIFT: u64 = 23;
+
+fn get_l2_entry(phys_addr: SupervisorPageAddr) -> MttL2EntryType {
+    let addr = phys_addr.bits();
+    let mtt_index = (addr & L2_ADDR_MASK) >> L2_ADDR_SHIFT;
+    let mtt_entry_addr = (get_mtt_base() + mtt_index) as *const u64;
+    assert!(mtt_index <= (8 * 1024 * 1024) - 8);
+    // Safety: The MTT is a 8MB contiguous region allocated by trusted platform-FW. The TSM is
+    // the only entity responsible for maintaining it, and we are using a 23-bit index into it
+    let value = unsafe { *mtt_entry_addr };
+    if let Some(l2_entry) = MttL2EntryType::from_raw(value) {
+        l2_entry
+    } else {
+        panic!("MTT L2 entry for {addr:x} was invalid (value:x)")
+    }
+}
+
+fn set_l2_entry(phys_addr: SupervisorPageAddr, entry: MttL2EntryType) {
+    let addr = phys_addr.bits();
+    let mtt_index = (addr & L2_ADDR_MASK) >> L2_ADDR_SHIFT;
+    assert!(mtt_index <= (8 * 1024 * 1024) - 8);
+    let mtt_entry_addr = (get_mtt_base() + mtt_index) as *mut u64;
```

Comment:
same comment as above w.r.t. assert.

Also a helper to generate `mtt_entry_addr`s from `phys_addr` would reduce duplication.

Reply:
Ack.